src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3089,15 +3101,151 @@ GenTree* Lowering::LowerHWIntrinsicTernaryLogic(GenTreeHWIntrinsic* node)
                     }
                 }
 
-                if (!condition->OperIsHWIntrinsic(NI_AVX512F_ConvertMaskToVector))
+                if (condition->OperIsHWIntrinsic(NI_AVX512F_ConvertMaskToVector))
                 {
-                    break;
+                    GenTree* tmp = condition->AsHWIntrinsic()->Op(1);
+                    BlockRange().Remove(condition);
+                    condition = tmp;
                 }
+                else if (!varTypeIsMask(condition))
+                {
+                    if (!condition->OperIsHWIntrinsic())
+                    {
+                        break;
+                    }
+
+                    GenTreeHWIntrinsic* cndNode = condition->AsHWIntrinsic();
+                    NamedIntrinsic      cndId   = cndNode->GetHWIntrinsicId();
+
+                    switch (cndId)
+                    {
+                        case NI_AVX_Compare:
```

Comment:
For this transformation to happen, where do we check if AVX512 is available CPU feature and this is valid to do?

Reply:
The very top of the method, we have `assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL));` because this is the handler explicitly for `vpternlogd/q` which is AVX512 only